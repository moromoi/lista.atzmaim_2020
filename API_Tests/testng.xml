<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd"><suite name="All Test Suite"><test verbose="2" preserve-order="true" name="D:/Devel_lista2020/lista.atzmaim_2020/API_AccountCreation_DDT_Tests/src/test/resources/input.xlsx"><classes/></test></suite>