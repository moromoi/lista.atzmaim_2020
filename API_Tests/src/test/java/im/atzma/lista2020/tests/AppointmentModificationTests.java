package im.atzma.lista2020.tests;

import org.approvaltests.Approvals;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.util.ArrayList;

public class AppointmentModificationTests extends TestBase {

    @Test(priority = 1)
    public void createAppointment() {

        int category_id = app.serviceCreationHelper().createCategory();
        int service_id = app.serviceCreationHelper().createService(category_id);
        int client_id = app.clientCreationHelper().createClient();
        app.appointmentHelper().createAppointment(client_id, service_id, category_id);

        Approvals.verify(app.appointmentHelper().getAppointmentList());
    }

    @Test(priority = 2)
    public void modifyAppointment() {
        int category_id = app.serviceCreationHelper().getCategoryID();
        int service_id = app.serviceCreationHelper().getServiceID();
        int client_id = app.clientCreationHelper().getClientID();
        app.clientCreationHelper().modificateClient(client_id);
        ArrayList<Integer> IDs = app.appointmentHelper().getAppointmentID();
        for (int i = 0; i < IDs.size(); i++) {
            int appointment_id = IDs.get(i);
            app.appointmentHelper().modifyAppointment(client_id, service_id, category_id, appointment_id);
        }

        Approvals.verify(app.appointmentHelper().getAppointmentList());

    }

    @Test(priority = 3)
    public void deleteAppointment() {
        app.appointmentHelper().deleteAppointment();
        Assert.assertEquals(app.appointmentHelper().getAppointmentString(), "[]");

    }
}
