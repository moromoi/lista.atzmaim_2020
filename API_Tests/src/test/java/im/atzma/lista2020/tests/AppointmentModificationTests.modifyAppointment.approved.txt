Server=nginx
#####
Content-Type=application/json; charset=utf-8
Transfer-Encoding=chunked
Connection=keep-alive
Vary=Accept-Encoding
Set-#####
X-Request-#####
Strict-Transport-Security=max-age=604800
X-Content-Type-Options=nosniff
X-Robots-Tag=noindex
Content-Security-Policy=upgrade-insecure-requests
X-Powered-By=php/4.3.10-16
Content-Encoding=gzip

[
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    },
    {
        "#####
        "name": "Test_changeClientsData_after",
        "#####
        "#####
        "total_price": "100",
        "phone": "0543666666",
        "note": "call one hour after",
        "client_#####
        "#####
        "status": "status - before",
        "off_time": null,
        "address": "Israel, Maron 8, Tel Aviv",
        "birthdate": "03-01",
        "durationEditable": null,
        "added_date": "2020-03-19 10:30",
        "services": [
            {
                "color": "#50e3c1",
                "price": "50",
                "duration": 60,
                "count": 1,
                "#####
                "name": "TestService"
            }
        ]
    }
]